### Importing libraries

In [ ]:
import pandas as pd

##### Create a dataframe, sales, from a csv file _supermarket\_sales.csv_

In [4]:
sm_sales = pd.read_csv("C:/Users/Jeth_Nico/Desktop/PythonProjects/AdvancedProjects/supermarket_sales.csv")

##### Checking **sm_sales** dataframe information using the `info()` method. Displaying the first 5 rows and all the columns of the dataframe using `head()` method.

In [5]:
sm_sales.info()
sm_sales.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1007 entries, 0 to 1006
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Invoice ID     1000 non-null   object 
 1   Branch         1000 non-null   object 
 2   City           1000 non-null   object 
 3   Customer type  1000 non-null   object 
 4   Gender         1000 non-null   object 
 5   Product line   1000 non-null   object 
 6   Unit price     1000 non-null   float64
 7   Quantity       1000 non-null   float64
 8   Tax 5%         1000 non-null   float64
 9   Total          1000 non-null   float64
 10  Date           1000 non-null   object 
 11  Time           1000 non-null   object 
 12  Payment        1000 non-null   object 
dtypes: float64(4), object(9)
memory usage: 102.4+ KB


Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total  \
0       Health and beauty       74.69       7.0  26.1415  548.9715   
1  Electronic accessories       15.28       5.0   3.8200   80.2200   
2      Home and lifestyle       46.33       7.0  16.2155  340.5255   
3       Health and beauty       58.22       8.0  23.2880  489.0480   
4       Sports and travel       86.31       7.0  30.2085  634.3785   

         Date   Time      Payment  
0  01/05/2019  13:08      Ewallet  
1  03/08/2019  10:29         Cash  
2  03/03/2019  13:23  Credit card  
3  01/27/2019  20:33      Ewallet  
4  02/08/2019  10:37      Ewallet

##### There are total of 1007 entries. All attributes entries has the same number of entries. This concludes that there are null or empty data's in the dataset

##### checking the total number of blank rows using `isnull()` and `sum()`

In [6]:
sm_sales.isnull().sum()

Invoice ID       7
Branch           7
City             7
Customer type    7
Gender           7
Product line     7
Unit price       7
Quantity         7
Tax 5%           7
Total            7
Date             7
Time             7
Payment          7
dtype: int64

##### Let's remove all nulls/blanks using the `dropna()` method and check again the dataset

In [7]:
sm_sales = sm_sales.dropna()
sm_sales.info()
sm_sales.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
Index: 1000 entries, 0 to 1006
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Invoice ID     1000 non-null   object 
 1   Branch         1000 non-null   object 
 2   City           1000 non-null   object 
 3   Customer type  1000 non-null   object 
 4   Gender         1000 non-null   object 
 5   Product line   1000 non-null   object 
 6   Unit price     1000 non-null   float64
 7   Quantity       1000 non-null   float64
 8   Tax 5%         1000 non-null   float64
 9   Total          1000 non-null   float64
 10  Date           1000 non-null   object 
 11  Time           1000 non-null   object 
 12  Payment        1000 non-null   object 
dtypes: float64(4), object(9)
memory usage: 109.4+ KB


Invoice ID       0
Branch           0
City             0
Customer type    0
Gender           0
Product line     0
Unit price       0
Quantity         0
Tax 5%           0
Total            0
Date             0
Time             0
Payment          0
dtype: int64